### PR TITLE
Move last remaining routes in main file to new pattern

### PR DESCRIPTION
**Pull request by matthewmcgarvey**
Related #2672
Related #2498

With this change, all routes will have been moved to the new setup.

### CODE REVIEW

File: src/invidious.cr [hunk 1]
```diff
@@ -154,8 +154,8 @@ if CONFIG.popular_enabled
   Invidious::Jobs.register Invidious::Jobs::PullPopularVideosJob.new(PG_DB)
 end
 
-connection_channel = Channel({Bool, Channel(PQ::Notification)}).new(32)
-Invidious::Jobs.register Invidious::Jobs::NotificationJob.new(connection_channel, CONFIG.database_url)
+CONNECTION_CHANNEL = Channel({Bool, Channel(PQ::Notification)}).new(32)
```
Comment (matthewmcgarvey) [inline src/invidious.cr:157]: Made a constant so that it could be referenced in the route file

File: src/invidious.cr [hunk 2]
```diff
@@ -399,92 +402,19 @@ Invidious::Routing.get "/s_p/:id/:name", Invidious::Routes::Images, :s_p_image
...
 Invidious::Routing.get "/vi/:id/:name", Invidious::Routes::Images, :thumbnails
 
+Invidious::Routing.get "/channel/:ucid/live", Invidious::Routes::Live, :check
+Invidious::Routing.get "/user/:user/live", Invidious::Routes::Live, :check
+Invidious::Routing.get "/c/:user/live", Invidious::Routes::Live, :check
+
 # API routes (macro)
 define_v1_api_routes()
+Invidious::Routing.get "/api/v1/auth/notifications", Invidious::Routes::API::V1::Authenticated, :notifications_get
+Invidious::Routing.post "/api/v1/auth/notifications", Invidious::Routes::API::V1::Authenticated, :notifications_post
```
Comment (matthewmcgarvey) [inline src/invidious.cr:412]: Terrible names on these but not what it could be instead

File: src/invidious/routes/api/v1/authenticated.cr [hunk 3]
```diff
@@ -397,4 +397,22 @@ module Invidious::Routes::API::V1::Authenticated
...
+  end
+
+  def self.notifications_post(env)
+    env.response.content_type = "text/event-stream"
+
+    topics = env.params.body["topics"]?.try &.split(",").uniq.first(1000)
+    topics ||= [] of String
+
+    create_notification_stream(env, topics, CONNECTION_CHANNEL)
+  end
```
Comment (SamantazFox) [inline src/invidious/routes/api/v1/authenticated.cr:417]: I think that you could merge those two in a single function, as the only difference is where the parameters are fetched from (`env.params.query` vs `env.params.body`).

File: src/invidious/routes/captcha.cr [hunk 4]
```diff
@@ -0,0 +1,8 @@
+module Invidious::Routes::Captcha
```
Comment (SamantazFox) [inline src/invidious/routes/captcha.cr:1]: could you please put that under the `Login` route, as it is related to Google log-in?

File: src/invidious/routes/live.cr [hunk 5]
```diff
@@ -0,0 +1,34 @@
+module Invidious::Routes::Live
```
Comment (SamantazFox) [inline src/invidious/routes/live.cr:1]: Same here: this route is related to channels, so it should go under `Invidious::Routes::Channels`

**Review by SamantazFox — CHANGES_REQUESTED** [review]

**Review by SamantazFox — APPROVED** [review]

**Pull request merged**